packages/frame-core/src/types.ts

```diff
@@ -40,38 +63,36 @@ export type FrameContext = {
     pfpUrl?: string;
   };
   location?: FrameLocationContext;
+  client: {
```

Comment:
I think this works.

consider optional `clientFid` so clients can self ID
`frameAdded` can be simplified to `added`
`notificationDetails` could be `notificationToken`

Reply:
we need the url there as well, `notificationDetails` is what we use in webhooks and the `addFrame` result so prefer to keep it for consistency